### DISCUSSION TITLE
Why would points::appendAttribute not like a vec3R attribute?

### DISCUSSION THREAD

**Discussion by cgolchert** [AcademySoftwareFoundation/openvdb#1605] [Q&A]
I have some 3rd party code that I'm trying to debug.

It has ALMOST exactly the points cookbook code except it uses _vec3R_ instead of _float_.

>   using Codec = openvdb::points::FixedPointCodec</*1-byte=*/false, openvdb::points::UnitRange>;
>   openvdb::points::TypedAttributeArray<**Vec3R**, Codec>::registerType();
>   openvdb::NamePair testAttribute = openvdb::points::TypedAttributeArray<**Vec3R**, Codec>::attributeType();

The above Vec3R breaks the 

>   openvdb::points::appendAttribute(grid->tree(), "test", testAttribute );

If I switch it to _float_ it becomes the cookbook example and works.  Can anyone tell me what is wrong?

**Comment by danrbailey** [comment]
Hi Chris, most of these compression schemes are designed primarily for reducing the size of float attributes (as opposed to double attributes), so I would recommend trying with Vec3S attributes. Otherwise, it might be helpful to see the error you’re running into?

**Comment by cgolchert** [comment]
I'm not sure why they were using a Vec3R.  I got it working with a nullcodec and found a few weird headers that had a typo.  I'll see if their stuff works with the Vec3S and a actual codec.  Thanks so much.
